src/common/com/intellij/plugins/haxe/lang/parser/HaxeGeneratedParserUtilBase.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017-2017 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.lang.parser;
+
+import com.intellij.lang.PsiBuilder;
+import com.intellij.lang.parser.GeneratedParserUtilBase;
+import com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypes;
+
+public class HaxeGeneratedParserUtilBase extends GeneratedParserUtilBase {
```

Comment:
This isn't going to work properly (not that the current code does).  Whitespace within the operators won't be noticed.  So...
Replace the file with this:
```java
/*
 * Copyright 2017-2017 Ilya Malanin
 * Copyright 2017 Eric Bishton
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.intellij.plugins.haxe;

import com.intellij.lang.PsiBuilder;
import com.intellij.lang.parser.GeneratedParserUtilBase;
import com.intellij.psi.tree.IElementType;

import static com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypes.*;

/**
 * Created by ebishton on 5/26/17.
 */
public class HaxeGeneratedParserUtilBase implements GeneratedParserUtilBase {

  private static boolean parseOperator(PsiBuilder builder_, int level_, IElementType target, IElementType... tokens) {
    final PsiBuilder.Marker marker_ = builder_.mark();
    for (IElementType token : tokens) {
      if (!nextTokenIsFast(builder_, token) || !consumeToken(builder_, token)) {
        marker_.rollbackTo();
        return false;
      }
    }
    marker_.collapse(target);
    return true;
  }
  
  public static boolean shiftRight(PsiBuilder builder_, int level_) {
    return parseOperator(builder_, level_, OSHIFT_RIGHT, OGREATER, OGREATER);
  }

  public static boolean shiftRightAssign(PsiBuilder builder_, int level_) {
    return parseOperator(builder_, level_, OSHIFT_RIGHT_ASSIGN, OGREATER, OGREATER, OASSIGN );
  }
  
  public static boolean unsignedShiftRight(PsiBuilder builder_, int level_) {
    return parseOperator(builder_, level_, OUNSIGNED_SHIFT_RIGHT, OGREATER, OGREATER, OGREATER);
  }
  
  public static boolean unsignedShiftRightAssign(PsiBuilder builder_, int level_) {
    return parseOperator(builder_, level_, OUNSIGNED_SHIFT_RIGHT_ASSIGN, OGREATER, OGREATER, OGREATER, OASSIGN);
  }
  
  public static boolean gtEq(PsiBuilder builder_, int level_) {
    return parseOperator(builder_, level_, OGREATER_OR_EQUAL, OGREATER, OASSIGN);
  }
}
```
The difference is the nextTokenIsFast() call, which makes sure that the next token is exactly the one you want without forwarding past the whitespace (and comments, etc.).  consumeToken() always forwards past whitespace.